site/app/libraries/Button.php

```diff
@@ -0,0 +1,77 @@
+<?php
```

Comment:
This feels like it'd be a model to me, albeit one not filled with DB information

Reply:
Yeah, I wasn't sure if that counted as model because it's not inheriting from AbstractModel. If PHP supported nesting classes I would have just put it inside NavigationView.

Reply:
Why not have it inherit from AbstractModel? You'd be able to get rid of all of your getter functions and just use the magic `__call` function to handle it?

Reply:
That's smart. I'll go change that